tests/test_operations.py

```diff
@@ -68,6 +75,22 @@ def test_stage_commit_push_changes__push_failure__raises_pygitops_error(
             )
 
 
+def test_feature_branch_context_manager__add_new_file_in_different_branches__clean_up_successfully(
```

Comment:
great! I think there are some other scenarios we would want to test as well.

what about changes where:
* an existing file is removed
* an existing file is modified

these test cases will help us ensure that your cleanup implementation does what we would like it to do!

Reply:
make sense! I'll add these test cases.